src/storage/src/source/postgres.rs

```diff
@@ -236,6 +238,12 @@ impl SourceReader for PostgresSourceReader {
             .expect("Postgres connection unexpectedly missing secrets");
 
         if active_read_worker {
+            let mut source_tables = HashMap::new();
+            let tables_iter = connection.details.tables.iter();
+            for (i, (table, table_cast)) in tables_iter.zip(connection.table_casts).enumerate() {
+                source_tables.insert(table.oid, (i + 1, table.clone(), table_cast));
```

Comment:
this `i+1` is terrifying, maybe place the index in the `table_casts` object as well?